### DISCUSSION TITLE
ESP32-S3 support

### DISCUSSION THREAD

**Discussion by Geoide** [Moddable-OpenSource/moddable#923] [General]
Hello.

What is the current support for esp32-s3? I have one but following the existing documentation, although I do load programs, I am not able to see anything in the xsbug console or interact with the board (for example, flashing an LED).

Thanks

**Comment by phoddie** [comment]
@Geoide – the ESP32-S3 port is working well, including xsbug support. There was [an issue](https://github.com/Moddable-OpenSource/moddable/issues/902) a while back, but it has since been resolved. I know of several developers that have succeeded in getting it to work. @mkellner may be able to provide some advice on what to check on your set-up.

**Reply by Geoide** [reply]
Thanks @phoddie 

And... is there some way to get data from de new usb OTG port?

Thanks.

**Comment by mkellner** [comment]
Hello @Geoide, There are quite a few esp32s3 modules wired in a bunch of different ways. If you've got the Espressif ESP32-S3-DevKitM (or equivalent), there are two ports at the bottom. Connect to the `UART` port for loading programs to the board and running xsbug.

You may have to set the UPLOAD_PORT environment variable to the device path for the esp32s3. For example, when I plug in my esp32s3 into my Mac, there is a new entry in /dev - `/dev/cu.usbserial-144340`. I would then `export UPLOAD_PORT=/dev/cu.usbserial-144340` before using `mcconfig -d -m -p esp32/esp32s3`.

**Comment by Geoide** [comment]
Thanks for your help @mkellner. 

I meant the new USB OTG to connect my old gamepad and get the data from it in order to use it like a remote control. I didn't refer to the usual usb port to comunicate with the computer.

**Reply by mkellner** [reply]
@geoide, I don't believe there is a project yet to build USB OTG modules.
